### ISSUE TITLE
Handle closed indices

### ISSUE THREAD

**Issue by pickypg** [elastic/support-diagnostics#335]
While looking at a diagnostic today, I noticed that `shards.json` and `cat_shards.txt` (at least) did not handle closed indices properly and thus simply returned errors.

There may be other APIs that choke for that reason, but those were the two files I wanted the most. Fortunately, `cat_indices.txt` worked.

Diagnostic: 7.1.2

**Comment by nemonster** [comment]
Not sure what you want me to do on this one? I don't actually _handle_ anything in the diag. I just write out what I get. If it comes back screwed up, that's what goes in file.

It seems like this should be a ticket for the Elastic devs...

**Comment by pickypg** [comment]
Indeed, although it looks like it resolves itself in later versions of Elasticsearch and there's no workaround prior to that.

I was expecting there to be a way to pass in the "index options" like Stack Monitoring does, but the `_cat/shards` API does not allow this and I couldn't remember how you got `shards.json` before looking just now (also `_cat/shards` using the JSON format for anyone wondering). Stack Monitoring overrides those options to ignore closed indices, which may still be relevant to some other APIs, but none that I found after a quick look, so just FYI:

https://github.com/elastic/elasticsearch/blob/25ffce9391b43552de469ce7bfaf1b38c4307af8/x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/indices/IndexStatsCollector.java#L61

I just tested `_cat/shards` with a `closed` index in 7.4.2 though and the API responds as you would hope:

```
index                    shard prirep state   docs   store ip            node
test                     0     p      STARTED              172.25.27.172 datanode2
test                     0     r      STARTED              172.25.150.23 datanode1
```

Closing this one out.
